android/library/src/main/java/com/liferay/mobile/screens/asset/display/interactor/AssetDisplayInteractor.java

```diff
@@ -72,7 +90,11 @@ protected String getIdFromArgs(Object... args) {
 		if (args.length > 1) {
 			cacheId = (long) args[1];
 		} else {
-			cacheId = (long) args[0];
+			if (args[0] instanceof Long) {
```

Comment:
we should move this ugly casting logic somewhere (I know there is something in the screenlet also)